web/api/v1/api.go

```diff
@@ -1007,10 +1007,10 @@ type RuleGroup struct {
 	// In order to preserve rule ordering, while exposing type (alerting or recording)
 	// specific properties, both alerting and recording rules are exposed in the
 	// same array.
-	Rules          []rule    `json:"rules"`
-	Interval       float64   `json:"interval"`
-	EvaluationTime float64   `json:"evaluationTime"`
-	LastEvaluation time.Time `json:"lastEvaluation"`
+	Rules               []rule    `json:"rules"`
+	Interval            float64   `json:"interval"`
+	EvaluationDuration  float64   `json:"evaluationTime"`
+	EvaluationTimestamp time.Time `json:"lastEvaluation"`
```

Comment:
These are part of a public API, so we can't change them. It's the other references that need updating.

Reply:
Yes, these are public APIs. However, `GetEvaluationDuration` in pkg `rules` is also a public API. Is it inappropriate to change only the field name `evaluationDuration` behind `GetEvaluationDuration`?

Might need more information.

Reply:
Our GO code is not subject to that restriction.

Reply:
Sorry, but I still not understand. 

The name `LastEvaluation ` and `EvaluationTime ` are also used in Thanos and Cortex and can be understood as a specification that can not be easily modified.  Is it unnecessary to modify the private field name without modifying the method signature?

Reply:
We're generally allowed to change all the internal Go APIs, but not the HTTP APIs.

Reply:
One suggestion is that it might be necessary to define an API specification for the prometheus ecosystem, some of which are common to a number of projects, and other than those who are well acquainted with Prometheus and its environs, others will have to confirm these details.

Reply:
> We're generally allowed to change all the internal Go APIs, but not the HTTP APIs.

But actually not sure, some PR is only for the GO-API which not be approved. And now many applications use the excellent prometheus library, and a specification may be required, such as the `k8s.io/apimachinery`.

Reply:
That's all well out of scope, and explicitly have no API currently as it'd be too much of a restriction on us.

Can you change the PR?

Reply:
Updated.

Reply:
Updated.